config/users-model.js

```diff
@@ -0,0 +1,24 @@
+const db = require('../database/dbConfig.js');
+
+module.exports = {
+  add,
+  findBy,
+  findById,
+};
+
+function findBy(filter) {
+  return db('users').where(filter);
+}
+
+async function add(user) {
+  const [id] = await db('users').insert(user);
+
+  return findById(id);
+}
+
+function findById(id) {
+  return db('users')
+    .select('id', 'username')
+    .where({ id })
+    .first();
```

Comment:
The design of this function with excluding passwords might make you think that you can't use your `findBy` helper function, but you _can_ call `select` after other queries:
```javascript
  return findBy({ id })
    .select('id', 'username')
    .first();
```